### PR TITLE
[R4R] fix pipecommit issue

### PR DESCRIPTION
**Pull request by forcodedancing**
### Description

This pr will fix an issue, i.e., some account root is not correctly set when committing mpt ,during pipecommit.

### Rationale

During pipecommit, the account root is populated as `dummyRoot` and will be corrected at several places. However, the previous code will skip some accounts during correction stage.  This will lead to commit error when committing mpt.
This fix will correct the root for `stateOjbect`, which's account root length is zero.

### Example

NA

### Changes

Notable changes: 
* change CorrectAccountRoots function

### CODE REVIEW

**Review by guagualvcha — APPROVED** [review]

**Review by guagualvcha — CHANGES_REQUESTED** [review]
It will keep as dummy root?

**Review by guagualvcha — DISMISSED** [review]

**Review by yutianwu — APPROVED** [review]

File: core/state/statedb.go [hunk 1]
```diff
@@ -989,7 +989,7 @@ func (s *StateDB) CorrectAccountsRoot(blockRoot common.Hash) {
 	if accounts, err := snapshot.Accounts(); err == nil && accounts != nil {
 		for _, obj := range s.stateObjects {
 			if !obj.deleted && !obj.rootCorrected && obj.data.Root == dummyRoot {
-				if account, exist := accounts[crypto.Keccak256Hash(obj.address[:])]; exist && len(account.Root) != 0 {
+				if account, exist := accounts[crypto.Keccak256Hash(obj.address[:])]; exist {
```
Comment (setunapo) [inline core/state/statedb.go:992]: This is the only place that changes the behavior, others changes in this PR is just code refine.
But how could `len(account.Root) == 0` ?

**Comment by setunapo** [comment]
Another suggestion, commit description should add more detail to tell how it is fixed, better to avoid `fix a bug`

**Comment by setunapo** [comment]
Another suggestion, commit description should add more detail to tell how it is fixed, better to avoid `fix a bug`

**Comment by forcodedancing** [comment]
> Another suggestion, commit description should add more detail to tell how it is fixed, better to avoid `fix a bug`

sure, let me refine it.

**Comment by forcodedancing** [comment]
> Another suggestion, commit description should add more detail to tell how it is fixed, better to avoid `fix a bug`

sure, let me refine it.

**Review by setunapo — DISMISSED** [review]

**Review by yutianwu — APPROVED** [review]

**Review by setunapo — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**